classes/requests/helpers/class-dintero-checkout-cart.php

```diff
@@ -360,16 +360,18 @@ public function get_pickup_points( $rate ) {
 	 * @return array
 	 */
 	public function get_shipping_item( $shipping_rate ) {
+		$shipping_cost = floatval( $shipping_rate->get_cost() );
+
 		$shipping_option = array(
 			'id'              => $shipping_rate->get_id(),
 			'line_id'         => $shipping_rate->get_id(),
-			'amount'          => self::format_number( $shipping_rate->get_cost() + $shipping_rate->get_shipping_tax() ),
+			'amount'          => self::format_number( $shipping_cost + $shipping_rate->get_shipping_tax() ),
 			'operator'        => '',
 			'description'     => '',
 			'title'           => html_entity_decode( $shipping_rate->get_label() ),
 			'delivery_method' => 'unspecified',
 			'vat_amount'      => self::format_number( $shipping_rate->get_shipping_tax() ),
-			'vat'             => ( empty( floatval( $shipping_rate->get_cost() ) ) ) ? 0 : self::format_number( $shipping_rate->get_shipping_tax() / $shipping_rate->get_cost() ),
+			'vat'             => $shipping_rate->get_cost() <= 0 ? 0 : self::format_number( $shipping_rate->get_shipping_tax() / $shipping_cost ),
```

Comment:
Suggestion: Here you are still using `$shipping_rate->get_cost()` when making the comparison. That should probably use the `$shipping_cost` variable now instead to prevent the same issue or a false positive here.